python/src/scipp/plot/model.py

```diff
@@ -0,0 +1,238 @@
+                    coord = data_array.coords[underlying_dim]
+                    coord = sc.Variable([dim],
+                                        values=coord.values,
+                                        variances=coord.variances,
+                                        unit=coord.unit,
+                                        dtype=sc.dtype.float64)
+                    coord_values = coord.values
+                else:
+                    coord = make_fake_coord(dim, dim_to_shape[dim] + 1)
+                    coord_values = coord.values
+                    if data_array.coords[dim].shape[-1] == dim_to_shape[dim]:
+                        coord_values = to_bin_centers(coord, dim).values
+                form = lambda val, pos: value_to_string(  # noqa: E731
+                    data_array.coords[dim].values[np.abs(coord_values - val).
+                                                  argmin()])
+                formatter.update({"linear": form, "log": form})
+
+            else:
+                coord = data_array.coords[dim].astype(sc.dtype.float64)
+
+        else:
+            # dim not found in data_array.coords
+            coord = make_fake_coord(dim, dim_to_shape[dim] + 1)
+
+        return coord, formatter
+
+    def get_axformatter(self, name, dim):
+        return self.axformatter[name][dim]
+
+    def get_coord_center_value(self, name, dim, ind):
+        return to_bin_centers(
+            self.data_arrays[name].coords[dim][dim, ind:ind + 2],
+            dim).values[0]
+
+    def get_data_names(self):
+        return list(self.data_arrays.keys())
+
+    def get_data_coord(self, name, dim):
+        return self.data_arrays[name].coords[dim]
+
+    def _select_bins(self, coord, dim, start, end):
+        """
+        Method to speed up the finding of bin ranges
+        """
+        bins = coord.shape[-1]
+        if len(coord.dims) != 1:  # TODO find combined min/max
+            return dim, slice(0, bins - 1)
+        # scipp treats bins as closed on left and open on right: [left, right)
+        first = sc.sum(coord <= start, dim).value - 1
+        last = bins - sc.sum(coord > end, dim).value
+        if first >= last:  # TODO better handling for decreasing
+            return dim, slice(0, bins - 1)
+        first = max(0, first)
+        last = min(bins - 1, last)
+        return dim, slice(first, last + 1)
+
+    def resample_data(self, array, rebin_edges):
+        """
+        Resample a DataArray according to new bin edges.
+        """
+        dslice = array
+        # Select bins to speed up rebinning
+        for dim in rebin_edges:
+            this_slice = self._select_bins(array.coords[dim], dim,
+                                           rebin_edges[dim][dim, 0],
+                                           rebin_edges[dim][dim, -1])
+            dslice = dslice[this_slice]
+
+        # Rebin the data
+        for dim, edges in rebin_edges.items():
+            dslice = sc.rebin(dslice, dim, edges)
```

Comment:
yes for the thick slices it would be nice to use value based slicing. I would do this in a follow-up though i think.
For resampling 2d images we need to use rebin.